apps/homework/models.py

```diff
@@ -147,6 +152,14 @@ def score(self):
             self.retrieve_score_and_status()
         return self.stored_score
 
+    # Our SAS urls will expire in 24 hours, so invalidate our cache automatically every 24 hours
+    @cached_property_with_ttl(ttl=60 * 24)
```

Comment:
If we have a very very long expiration I don't think we need cached property?

Reply:
How would Chagrade know to retrieve an updated SAS url? Even if the SAS url expires in a week, when that comes, how would Chagrade ever be able to retrieve an updated SAS? It seems to me that a long cached property was the best way to handle this, but if there's a better way to handle this I'm all down for switching to that.

Reply:
It won't need to if expiry is 100 years from now, or what have you. I believe, at a minimum, the highest expiry we could set is 2038?

Reply:
👍